Source/Core/Core/ConfigLoaders/NetPlayConfigLoader.cpp

```diff
@@ -146,6 +148,11 @@ class NetPlayConfigLayerLoader final : public Config::ConfigLayerLoader
     }
 #endif
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+    Config::SetBaseOrCurrent(Config::RA_HARDCORE_ENABLED, m_settings.enable_hardcore);
+    AchievementManager::GetInstance().SetHardcoreMode();
```

Comment:
`AchievementManager::SetHardcoreMode` uses `m_client`, which is set in `AchievementManager::Init`. Is there any guarantee that `AchievementManager::Init` has been called at this point?

Reply:
After taking a second look: If `Init` hasn't been called, `m_client` will be `nullptr`, and `rc_client_set_hardcore_enabled` does return early if the client is `nullptr`. So there isn't actually a safety issue.

Reply:
Yep, already handled.